yarn-project/world-state/src/native/world_state_version.ts

```diff
@@ -0,0 +1,35 @@
+import { EthAddress } from '@aztec/circuits.js';
+import { BufferReader, serializeToBuffer } from '@aztec/foundation/serialize';
+
+import { readFile, writeFile } from 'fs/promises';
+
+export class WorldStateVersion {
+  constructor(readonly version: number, readonly rollupAddress: EthAddress) {}
+
+  static async readVersion(filename: string) {
+    const expectedLength = WorldStateVersion.empty().toBuffer().length;
+
+    const versionData = await readFile(filename).catch(() => undefined);
+    if (versionData === undefined || versionData.length != expectedLength) {
+      return undefined;
+    }
+    return WorldStateVersion.fromBuffer(versionData);
+  }
+
+  public async writeVersionFile(filename: string) {
+    await writeFile(filename, this.toBuffer());
+  }
```

Comment:
Should we use a string representation instead, so the file is human-readable?

Reply:
Yeah, I agree